lib/ImportFromC/HeaderToModel.cpp

```diff
@@ -608,57 +597,52 @@ bool DeclVisitor::VisitFunctionDecl(const clang::FunctionDecl *FD) {
         return false;
       }
 
-      auto RegisterID = model::Register::fromCSVName(*ReturnValue,
-                                                     Model->Architecture());
-      if (RegisterID == model::Register::Invalid) {
+      auto Location = model::Register::fromCSVName(*Register,
+                                                   Model->Architecture());
+      if (Location == model::Register::Invalid) {
         revng_log(Log, "Unsupported register location");
         return false;
       }
 
-      auto &ReturnValueReg = ReturnValuesInserter.emplace(RegisterID);
+      auto &ReturnValueReg = ReturnValuesInserter.emplace(Location);
       ReturnValueReg.Type() = std::move(RetType);
     }
 
     auto ArgumentsInserter = TheRawFunctionType.Arguments().batch_insert();
     for (unsigned I = 0, N = FD->getNumParams(); I != N; ++I) {
       auto ParamDecl = FD->getParamDecl(I);
-      if (ParamDecl->hasAttr<AnnotateAttr>()) {
-        auto Annotate = std::find_if(begin(ParamDecl->getAttrs()),
-                                     end(ParamDecl->getAttrs()),
-                                     [&](Attr *Attribute) {
-                                       return isa<AnnotateAttr>(Attribute);
-                                     });
-        auto Loc = getLoc(cast<AnnotateAttr>(*Annotate)->getAnnotation());
-        if (not Loc or Loc->empty()) {
-          revng_log(Log,
-                    "Parameters should have attribute annotate with reg or "
-                    "stack");
-          return false;
-        }
-
-        // TODO: Handle stack location.
-        if (*Loc == "stack") {
-          revng_log(Log, "We don't support parameters on stack for now");
+      std::optional Register = parseStringAnnotation(*ParamDecl, RegAnnotation);
+      if (not Register.has_value()) {
+        std::optional Stack = parseStringAnnotation(*ParamDecl,
+                                                    StackAnnotation);
```

Comment:
I think in RFT we also allow only a single stack argument and it must have a pointer type in C. Isn't it?

Reply:
Yep, sounds like a good assertion, thanks.

Reply:
It felt cheesy to extract information just so I can assert it, so I left a TODO instead.

Reply:
No offence, but it sounds silly. The very moment I read "TODO: don't forget to assert that it is always to assert" I think: "Why haven't you asserted then?".

So please, just assert it.

Reply:
Done.